### DISCUSSION TITLE
Unable to retrieve minified file of livewire.js

### DISCUSSION THREAD

**Discussion by wize-wiz** [livewire/livewire#6838] [7. Bugs (archived)] [answered]
I've wondered why `livewire.js` is not minified when changing the environment to **production**. 

I figured that while `Livewire\Mechanisms\FrontendAssets\FrontendAssets::returnJavaScriptAsFile()` has [this](https://github.com/livewire/livewire/blob/c6779a499bbc786b18c043ec633fd98927d740be/src/Mechanisms/FrontendAssets/FrontendAssets.php#L71) line `return Utils::pretendResponseIsFile(__DIR__.'/../../../dist/livewire.js');` hardcoded, it can **never** return a minified file.

Is the [route callback](https://github.com/livewire/livewire/blob/c6779a499bbc786b18c043ec633fd98927d740be/src/Mechanisms/FrontendAssets/FrontendAssets.php#L47C14-L47C28) overwritten in a place somewhere else I haven't found yet? I can find no mention of `min.js` or `FrontendAssets::javaScriptRoute` being set anywhere else.

In the `livewire/dist` folder, there is a minified version available, of course :)

**Comment by lrljoe** [comment]
Not as far as I'm aware!  I think this is perhaps just an oversight in terms of whether minified or unminified are returned, worth a PR perhaps?

I actually use similar injection methodology in other packages, but with minified versions, so it's definitely do-able.

For Livewire core scripts, I let Nginx take care of gzipping the JS, which reduces the unminified version down to 80kb.

**Reply by wize-wiz** [reply]
The minified version is at 45kb :)

**Comment by edjeavons** [comment]
Changing `src/Mechanisms/FrontendAssets/FrontendAssets.php` as follows makes the regular asset URL (https://example.com/livewire/livewire.js) return the minified version without any wider changes.
```
public function returnJavaScriptAsFile()
{
    return Utils::pretendResponseIsFile(__DIR__.'/../../../dist/livewire.min.js');
}
```
It offers a significant saving (112KB vs. 261KB) so is there a practical reason for not returning the minified file by default, regardless in the environment?

**Comment by ChimpGamer** [comment]
I'm also wondering this :)

**Comment by wize-wiz** [comment]
@edjeavons @lrljoe I don't have any idea what this new thing called "Issue-less Issue" is on github. These "discussions to PR without Issues" are quite horrible and I just refuse to get involved, somehow :)

Until it will magically work ootb and without changing the base code, overwrite the `livewire/livewire.js` route in `web.php` returning the minified version. Just make sure this route gets registered after the packages have been registered.

```
Route::get('livewire/livewire.js', function() {
    return \Livewire\Drawer\Utils::class::pretendResponseIsFile(
        base_path('vendor/livewire/livewire/dist/livewire.min.js')
    );
});
```

**Reply by edjeavons** [reply]
Thanks for the tip, that has the desired effect.

I still hope this becomes the default at some point though.

**Comment by sammyaxe** [comment]
how come this is still not addressed? For me overriding route doesn't seem to help.

**Reply by piotrku** [reply]
Working for me.
Maybe you need to clear the route cache?
`php artisan route:clear`

**Comment by joshhanley** [comment, marked as answer]
I've submitted a PR #7928 with a fix for this.

**Reply by edjeavons** [reply]
Thank you
